compiler/rustc_middle/src/ty/layout.rs

```diff
@@ -1856,6 +1856,29 @@ pub enum SizeSkeleton<'tcx> {
         /// depending on one, with regions erased.
         tail: Ty<'tcx>,
     },
+
+    /// A value whose size would be statically known after monomorphization, but may not yet be
+    /// knowable. The size must either be exactly one instance of a type, or an array of instances
+    /// of a type, because this variant isn't aware of other layout constraints such as padding.
+    ///
+    /// This can be useful for comparing two types which can be known to be the same size because,
+    /// though individually their sizes aren't known, they can be easily proven to have the same
+    /// size. For instance, checking whether two arrays of the same type parameter have the same
+    /// size (even though the type parameter's size may not yet be known), or for comparing the
+    /// size of a value of a type parameter with a repr(transaprent) type which contains a value of
+    /// the same type parameter.
+    ///
+    /// This is, however, currently very limited and conservative. For instance, two arrays of the
+    /// same type, where:
+    ///  * one's size is a const generic parameter
+    ///  * the other's is an const value which happens to be equal to that const generic parameter
+    /// will not be considered equal.
+    UnresolvedConstant {
```

Comment:
Maybe something like `RepeatedTy` or so would make more sense? The current name doesn't really convey a lot of meaning.